### ISSUE TITLE
Unnecessary errors with 'field is required'

### ISSUE THREAD

**Issue by jonathanzhang53** [openMF/community-app#3046]
## Description
When you get an error, you always get a 'field is required' error, even if all the fields are filled in. This is given along with the actual error.

## Steps to Reproduce
[THIS CAN BE REPRODUCED IN MANY PARTS OF THE APP WITH INPUT FIELDS]
One Example:
- Open [https://demo.openmf.org](url)
- Click Accounting
![image](https://user-images.githubusercontent.com/41968151/48308601-70e34900-e52e-11e8-8699-dc0bfb8e0b35.png)
- Click Add Journal Entries
![image](https://user-images.githubusercontent.com/41968151/48308603-7c367480-e52e-11e8-91f7-15be8e467972.png)
- Enter all fields, but make credit 100 and debit 101 (just make them unequal so error is thrown) (see screenshot below)
- Submit and see credit/debit error but also an unnecessary 'field is required' error (since all fields are filled)

## Expected Behaviour
No field is required error if all fields are filled in with only the relevant errors displayed

## Actual Behaviour
Shows actual error AND 'field is required' error

## Settings
- Mifos X version: Release Version: 18.03.01.RELEASE
- Browser used: Chrome

## Screenshots, if any
![image](https://user-images.githubusercontent.com/41968151/48294659-479dbc80-e44b-11e8-82b1-eccde2408bc0.png)

**Comment by Rupeshiya** [comment]
@jonathanzhang53  I found that there is error on handling POST request of api .That's why unable to post the data .

**Comment by jonathanzhang53** [comment]
@Rupeshiya Does that relate to the error handling? Not sure what you are referring to.

**Comment by ShruthiRajaram** [comment]
@Rupeshiya , getting field is required message when all fields are entered is not good (Displaying what went wrong is enough).@jonathanzhang53 agreed. this needs to be handled in ApiValidationDirective

**Comment by Rupeshiya** [comment]
Yes @jonathanzhang53  @ShruthiRajaram  this needs to be handled in ApiValidationDirective.

**Comment by debenick17** [comment]
Can I work on this issue?

**Comment by loveloud** [comment]
Hello, I would like to fix this, any guidance on where I can start?

**Comment by santoshconflux** [comment]
@Hackingforlife , Go to accounting section and follow the steps given in description.

**Comment by loveloud** [comment]
@santoshconflux,  I'm on it.

**Comment by loveloud** [comment]
@santoshconflux, Hello, I was able to reproduce the error but then I have been trying to troubleshoot, but I'm still not sure where to find the API Validation Directive code, I have not used Angular before. I'm stuck here.

**Comment by santoshconflux** [comment]
@Hackingforlife , Please send mail to dev mailing list. or enquire in Gitter.

**Comment by loveloud** [comment]
Alright.Thanks

**Comment by loveloud** [comment]
@santoshconflux, Kindly help me with a link to the mailing list.Thanks

**Comment by santoshconflux** [comment]
@Hackingforlife ,To subscribe, send a blank email to dev-subscribe@fineract.apache.org.

**Comment by loveloud** [comment]
@santoshconflux , Thanks

**Comment by loveloud** [comment]
@santoshconflux ,i did send the email,but looks like this email address doesnot work anymore.I was not successful
